### PR TITLE
Add Authentication to project

### PR DESCRIPTION
**Pull request by RakshaPawar108**
Changes to check for since the previous PR

- Auth context and reducer added
- Service for login and signup (API calls)
- Signup feature added
- Login feature added
- Login with guest credentials added
- Check for all fields filled and alert if not
- Check for password and confirm password fields
- Navbar changed post login and signup
- Logout feature

**Deploy Preview Link Given Below**

### CODE REVIEW

File: src/pages/Authentication/Login/Login.jsx [hunk 1]
```diff
@@ -1,21 +1,76 @@
...
+          },
+        });
+
+        alert("Logged In successfully");
+        navigate("/");
+      } else {
+        alert("ERROR IN LOGGING IN!");
+      }
+    } catch (err) {
+      console.log(err);
```
Comment (RahullRawat) [inline src/pages/Authentication/Login/Login.jsx]: You can remove these `console.log`

File: src/pages/Authentication/Signup/Signup.jsx [hunk 2]
```diff
@@ -1,35 +1,130 @@
...
+                user: response.data.createdUser,
+                token: response.data.encodedToken,
+              },
+            });
+
+            alert("Signed Up successfully");
+            navigate("/");
+          }
+        } catch (err) {
+          console.log(err);
```
Comment (RahullRawat) [inline src/pages/Authentication/Signup/Signup.jsx]: You can remove these `console.log`

File: src/services/provideAuth.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      const response = await axios.post("/api/auth/signup", user);
+      if (response.status === 201) {
+        localStorage.setItem("token", response.data.encodedToken);
+        localStorage.setItem("user", JSON.stringify(response.data.createdUser));
+      } else {
+        throw new Error();
+      }
+      return response;
+    } catch (err) {
+      console.log(err);
```
Comment (RahullRawat) [inline src/services/provideAuth.jsx]: You can remove these `console.log`

File: src/context/auth-context.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+
+import { authReducer } from "../reducers";
+
+const AuthContext = createContext(null);
+
+const AuthProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(authReducer, {
```
Comment (RahullRawat) [inline src/context/auth-context.jsx]: Just a small suggestion, as we are going to have multiple states and dispatch so to differentiate, you can name it something as `[authState, authDispatch]`

**Review by RahullRawat — APPROVED** [review]
- Auth functionality is working as intended.
- Very clean and well formatted code with good naming conventions makes reviewing easy

File: src/context/auth-context.jsx (hunk 4 above)
Comment (RakshaPawar108) [reply src/context/auth-context.jsx]: Done. Thanks for the suggestion!

**Review by Kevin-Solomon — APPROVED** [review]

**Pull request merged**